docs/book/src/developers/development.md

```diff
@@ -559,6 +559,7 @@ You can optionally set the following variables:
 | `TEST_WINDOWS`                  | Build a cluster that has Windows worker nodes.          |
 | `REGISTRY`                  | Registry to push any custom k8s images or cloud provider images built.                                                                     |
 | `CLUSTER_TEMPLATE`          | Use a custom cluster template. By default, the script will choose the appropriate cluster template based on existing environment variables. |
+| `CLUSTER_TEMPLATE_LINK` | Use a custom cluster template link outside of CAPZ repo. |
```

Comment:
what do you think about reusing the `CLUSTER_TEMPLATE` variable instead of adding a new variable? 

That way we can just curl the template in the makefile and apply it directly if the template isn't a local file:

```
  if [ -f $(TEMPLATES_DIR)/$(CLUSTER_TEMPLATE) ]; then \
      $(ENVSUBST) < "$(TEMPLATES_DIR)/$(CLUSTER_TEMPLATE)" | kubectl apply -f -; \
else curl --retry $(CURL_RETRIES) $(CLUSTER_TEMPLATE) | $(ENVSUBST) | kubectl apply -f; - \
  fi
```

We could even add the option for a valid local file path that's not in the templates dir if we wanted.

Reply:
> We could even add the option for a valid local file path that's not in the templates dir if we wanted.

I started to look at this for some e2e tests for sig-windows gmsa and this would helpful.  Right now we can't overload the `TEMPLATES_DIR=` https://github.dev/kubernetes-sigs/cluster-api-provider-azure/blob/c08e5e631bdb1831bcaf698525f3c423e48a706c/Makefile#L45-L46

so are limited to templates in the template folder here.

Reply:
Makes sense. I have updated the code.